app/shop.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import annotations
+
+import dataclasses
+
+
+@dataclasses.dataclass
+class Shop:
+    shop_name: str
+    shop_location: list[int]
+    shop_prices: dict[str, float]
+
+    @staticmethod
+    def create_shop(shops: dict) -> list[Shop]:
+        shop_list = []
+        for shop in shops:
+            shop_list.append(
+                Shop(
+                    shop["name"],
+                    shop["location"],
+                    shop["products"]
+                )
+            )
+        return shop_list
+
+    def sell_products(
+            self,
+            product_cart: dict,
+            print_indicator: bool = True
+    ) -> float:
```

Comment:
but how to make additional function? Because idk how to avoid repeating in output

Reply:
```
from __future__ import annotations

from typing import Any
from dataclasses import dataclass


@dataclass
class Shop:
    shop_name: str
    shop_location: list[int]
    shop_prices: dict[str, float]

    @staticmethod
    def create_shop(shops: dict[str, Any]) -> list[Shop]:
        shop_list = [
            Shop(shop["name"], shop["location"], shop["products"])
            for shop in shops
        ]
        return shop_list

    def get_product_cost(
        self, name: str, count: int
    ) -> float:
        return count * self.shop_prices[name]

    def get_product_costs(
        self,
        product_cart: dict[str, int],
    ) -> float:
        return sum(
            self.get_product_cost(name, count)
            for name, count in product_cart.items()
        )

    def sell_products(
        self,
        product_cart: dict[str, int],
    ) -> float:

        print("You have bought: ")

        total_cost = 0
        for product, count in product_cart.items():
            product_cost = self.get_product_cost(product, count)
            total_cost += product_cost

            print(f"{count} {product}s for {product_cost} dollars")

        print(f"Total cost is {total_cost} dollars")

        return total_cost

```
I think maybe something like this. Also, maybe we can remove calculations for total cost use self.get_products_costs()
It is hard quickly find best solution how to avoid boolean flag